test/dashboard/index.spec.js

```diff
@@ -25,4 +41,251 @@ describe("dashboard", () => {
     expect(dashboardWithOptions.color).to.equal("red");
     expect(dashboardWithOptions.minimal).to.be.true;
   });
+
+  describe("set* methods", () => {
+    let dashboard;
+
+    beforeEach(() => {
+      dashboard = new Dashboard();
+    });
+
+    describe("setData", () => {
+      const dataArray = [{
+        type: "progress",
+        value: 0.57
+      }, {
+        type: "operations",
+        value: "IDLE"
+      }];
+
+      it("can setData", () => {
+        expect(dashboard.setData(dataArray)).to.not.throw;
+      });
+    });
+
+    describe("setOperations", () => {
+      const data = {
+        value: "IDLE"
+      };
+
+      it("can setOperations", () => {
+        expect(dashboard.setOperations(data)).to.not.throw;
+        expect(dashboard.operations.setContent).to.have.been.calledWith(data.value);
+      });
+    });
+
+    describe("setStatus", () => {
+      const data = {
+        value: "Success"
+      };
+
+      it("can setStatus", () => {
+        expect(dashboard.setStatus(data)).to.not.throw;
+        expect(dashboard.status.setContent)
+          .to.have.been.calledWith(`{green-fg}{bold}${data.value}{/}`);
+      });
+
+      it("should display a failed status on build failure", () => {
+        data.value = "Failed";
+
+        expect(dashboard.setStatus(data)).to.not.throw;
+        expect(dashboard.status.setContent)
+          .to.have.been.calledWith(`{red-fg}{bold}${data.value}{/}`);
+      });
+
+      it("should display any other status string without coloring", () => {
+        data.value = "Unknown";
+
+        expect(dashboard.setStatus(data)).to.not.throw;
+        expect(dashboard.status.setContent)
+          .to.have.been.calledWith(`{bold}${data.value}{/}`);
+      });
+    });
+
+    describe("setProgress", () => {
+      const data = {
+        value: 0.57
+      };
+
+      it("can setProgress", () => {
+        expect(dashboard.setProgress(data)).to.not.throw;
+        expect(dashboard.progressbar.setProgress).to.have.been.calledOnce;
+        expect(dashboard.progressbar.setContent).to.have.been.called;
+      });
+
+      it(`should call progressbar.setProgress twice if not in minimal mode
+      and percent is falsy`, () => {
+        data.value = null;
+
+        expect(dashboard.setProgress(data)).to.not.throw;
```

Comment:
GLOBAL: 

These have to be of the format of "something to call", so:

```js
expect(() => { dashboard.setProgress(data); }).to.not.throw;
```

unless `dashboard.setProgress(data)` returns a _function to execute_, which I don't believe it does.

Reply:
Note: I was adding comments to everything I saw as "Same comment here.", but there are a lot, so just assume I did that 😛